frontend/vite.config.ts

```diff
@@ -1,37 +1,71 @@
 /* eslint-disable import/no-extraneous-dependencies */
 /// <reference types="vitest" />
 /// <reference types="vite-plugin-svgr/client" />
-import { defineConfig } from "vite";
+import { defineConfig, loadEnv } from "vite";
 import viteTsconfigPaths from "vite-tsconfig-paths";
 import svgr from "vite-plugin-svgr";
 import { vitePlugin as remix } from "@remix-run/dev";
 
-export default defineConfig(() => ({
-  plugins: [
-    !process.env.VITEST &&
-      remix({
-        future: {
-          v3_fetcherPersist: true,
-          v3_relativeSplatPath: true,
-          v3_throwAbortReason: true,
+export default defineConfig(({ mode }) => {
+  const {
+      VITE_BACKEND_HOST = "127.0.0.1:3000",
+      VITE_USE_TLS = "false",
+      VITE_FRONTEND_PORT = "3001",
+      VITE_INSECURE_SKIP_VERIFY = "false",
+      VITE_WATCH_USE_POLLING = "false",
+  } = loadEnv(mode, process.cwd());
+
+  const USE_TLS = VITE_USE_TLS === "true";
+  const INSECURE_SKIP_VERIFY = VITE_INSECURE_SKIP_VERIFY === "true";
+  const PROTOCOL = USE_TLS ? "https" : "http";
+  const WS_PROTOCOL = USE_TLS ? "wss" : "ws";
+
+  const API_URL = `${PROTOCOL}://${VITE_BACKEND_HOST}/`;
+  const WS_URL = `${WS_PROTOCOL}://${VITE_BACKEND_HOST}/`;
+  const FE_PORT = Number.parseInt(VITE_FRONTEND_PORT, 10);
+  return {
+    plugins: [
+      !process.env.VITEST &&
+        remix({
+          future: {
+            v3_fetcherPersist: true,
+            v3_relativeSplatPath: true,
+            v3_throwAbortReason: true,
+          },
+          appDirectory: "src",
+          ssr: false,
+        }),
+      viteTsconfigPaths(),
+      svgr(),
+    ],
+    server: {
+      port: FE_PORT,
```

Comment:
If default it is the default port that you want to fix (from `5173` to `3000`), all you need to set is [`server.port`](https://v3.vitejs.dev/config/server-options.html#server-port). I don't think we are utilizing anything proxy-related here (it is a [dev-only setting](https://v3.vitejs.dev/config/server-options.html#server-proxy))

Reply:
right exactly--in dev mode, the FE now makes requests to `localhost:3001/api/...` and the proxy passes them to `localhost:3000/api/...`

That's a bit more like prod, where both FE and BE are at the same address